### PR TITLE
dockle: init at 0.3.1

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Motivation for this change

Adds dockle scanner to nixpkgs at `0.3.1`

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Result of `nixpkgs-review pr 112906` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dockle</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112906` at 27ac26cc run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/34ee5baf772fd3d2661247621ec9a61d">dockle</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>error: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-array.md">no-flags-array</a>

`buildFlagsArray` is only intended to be used in Bash, not as a Nix attribute.

Near pkgs/development/tools/dockle/default.nix:19:3:
```
   |
19 |   buildFlagsArray = [
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/dockle/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  buildFlagsArray = [
+    "-ldflags="
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+  ];
+
+  preCheck = ''
+    # Remove tests that use networking
+    rm -r pkg/scanner/scan_test.go pkg/utils/fetch_test.go
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dockle/default.nix:28]:
```suggestion
    rm pkg/scanner/scan_test.go pkg/utils/fetch_test.go
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 112906` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dockle</li>
  </ul>
</details>

**Pull request merged**